File: FlashCards.Business/DTOs/StudyCardDTO.cs
using FlashCards.Core.Enums;

namespace FlashCards.Business.DTOs
{
    public class StudyCardDTO
    {
        public Guid Id { get; set; }
        public Guid DeckId { get; set; }
        public string FrontText { get; set; }
        public string BackText { get; set; }
        public Language? Language { get; set; }
        public GrammaticalGender? Gender { get; set; }
        public string? Plural { get; set; }
        public string? Pronunciation { get; set; }
        public QuestionType QuestionType { get; set; }
        public bool CheckFrontText { get; set; } = true;
        public ICollection<string> PossibleAnswers { get; set; } = new List<string>();
    }
}

File: german-cards-pwa/src/app/models/study-card-dto.ts
import { CardLanguage } from './card-language';
import { GrammaticalGender } from './grammatical-gender';
import { QuestionType } from './question-type';

export interface StudyCardDto {
  id: string;
  deckId: string;
  frontText: string;
  backText: string;
  displayedBackText: string;
  language?: CardLanguage | null;
  gender?: GrammaticalGender | null;
  plural?: string | null;
  pronunciation?: string | null;
  questionType: QuestionType;
  checkFrontText: boolean;
  possibleAnswers: string[];
}
